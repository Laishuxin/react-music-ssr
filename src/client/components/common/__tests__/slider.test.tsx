import React from 'react'
import { render } from '@testing-library/react'
import Slider from '../slider'
describe('slider', () => {
  it('should render the correct slider with default props', () => {
    const { getByTestId } = render(
      <Slider
        data-testid='test-slider'
        dataSource={[
          { key: 1, value: 1 },
          { key: 2, value: 2 },
        ]}
        render={item => <span>{item.value}</span>}
      />,
    )
    const element = getByTestId('test-slider')
    expect(element).toBeInTheDocument()
    expect(element).toMatchSnapshot('snapshot-slider')
  })
})
