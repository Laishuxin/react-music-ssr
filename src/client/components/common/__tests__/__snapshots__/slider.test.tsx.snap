// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`slider should render the correct slider with default props: snapshot-slider 1`] = `
<div
  class="relative flex justify-center box-border w-full h-full bg-white"
  data-testid="test-slider"
>
  <div
    class="slider-container relative z-10 rounded-lg top-2 w-full mx-4 h-full overflow-hidden swiper-container-initialized swiper-container-horizontal"
  >
    <div
      class="swiper-wrapper"
      style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"
    >
      <div
        class="swiper-slide swiper-slide-duplicate swiper-slide-duplicate-active"
        data-swiper-slide-index="1"
      >
        <span>
          2
        </span>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="0"
      >
        <span>
          1
        </span>
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="1"
      >
        <span>
          2
        </span>
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="0"
      >
        <span>
          1
        </span>
      </div>
    </div>
    <div
      class="swiper-pagination swiper-pagination-bullets"
    />
    <span
      aria-atomic="true"
      aria-live="assertive"
      class="swiper-notification"
    />
  </div>
</div>
`;
